### PR TITLE
decompile a portion of field_player_avatar.s

### PR DESCRIPTION
**Pull request by atasro2**

### CODE REVIEW

File: src/field_player_avatar.c [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+        return 0;
+}
+
+u8 sub_808B238(s16 x, s16 y, u8 direction)
+{
+    if (FlagGet(FLAG_SYS_USE_STRENGTH))
+    {
+        u8 eventObjectId = GetEventObjectIdByXY(x, y);
+
+        if (eventObjectId != 16)
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Combine if statements

File: src/field_player_avatar.c [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+    else
+    {
+        return 0;
+    }
+}
+
+bool8 ShouldJumpLedge(s16 x, s16 y, u8 z)
+{
+    if (GetLedgeJumpDirection(x, y, z) != 0)
+        return 1;
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: return TRUE and FALSE

File: src/field_player_avatar.c [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+}
+
+bool8 sub_808B1BC(s16 x, s16 y, u8 direction)
+{
+    if ((gPlayerAvatar.flags & PLAYER_AVATAR_FLAG_SURFING)
+     && MapGridGetZCoordAt(x, y) == 3
+     && GetEventObjectIdByXYZ(x, y, 3) == 16)
+    {
+        sub_808C750(direction);
+        return 1;
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: return TRUE and FALSE

File: src/field_player_avatar.c [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+            else
+            {
+                return;
+            }
+        }
+    }
+
+    if (gPlayerAvatar.flags & PLAYER_AVATAR_FLAG_SURFING)
+    {
+		// speed 2 is fast, same speed as running
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Fix tabs

File: src/field_player_avatar.c [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+void PlayerNotOnBikeTurningInPlace(u8 direction, u16 heldKeys)
+{
+    PlayerTurnInPlace(direction);
+}
+
+void PlayerNotOnBikeMoving(u8 direction, u16 heldKeys)
+{
+    u8 r0 = CheckForPlayerAvatarCollision(direction);
+
+    if(r0 != 0)
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Space between if and (

File: src/field_player_avatar.c [hunk 6]
```diff
@@ -0,0 +1,661 @@
...
+_0808AA94:\n\
+	pop {r4-r6}\n\
+	pop {r1}\n\
+	bx r1");
+}
+#endif
+
+void npc_clear_strange_bits(struct EventObject *eventObj)
+{
+	eventObj->inanimate = 0;
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Fix tab

File: src/field_player_avatar.c [hunk 7]
```diff
@@ -0,0 +1,609 @@
...
+    s16 x, y;
+    struct EventObject *playerEventObj = &gEventObjects[gPlayerAvatar.eventObjectId];
+
+    x = playerEventObj->currentCoords.x;
+    y = playerEventObj->currentCoords.y;
+    MoveCoords(direction, &x, &y);
+    return sub_808B164(playerEventObj, x, y, direction, MapGridGetMetatileBehaviorAt(x, y));
+}
+
+u8 CheckForEventObjectCollision(struct EventObject *a, s16 x, s16 y, u8 direction, u8 e)
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Single return pattern

File: src/field_player_avatar.c [hunk 8]
```diff
@@ -0,0 +1,609 @@
...
+    if (collision == 0)
+    {
+        if (CheckForRotatingGatePuzzleCollision(direction, x, y))
+            return 8;
+        check_acro_bike_metatile(x, y, e, &collision);
+    }
+    return collision;
+}
+
+u8 sub_808B164(struct EventObject *a, s16 x, s16 y, u8 direction, u8 e)
```
Comment (Diegoisawesome) [inline src/field_player_avatar.c]: Single return pattern

File: src/field_player_avatar.c [hunk 9]
```diff
@@ -0,0 +1,609 @@
...
+                MovePlayerAvatarUsingKeypadInput(direction, newKeys, heldKeys);
+                PlayerAllowForcedMovementIfMovingSameDirection();
+            }
+        }
+    }
+}
+
+bool8 TryInterruptEventObjectSpecialAnim(struct EventObject *playerEventObj, u8 direction)
+{
+    #ifdef NONMATCHING
```
[the remark this answers is not in the document]
Comment (atasro2) [reply src/field_player_avatar.c]: where should i try a return var ?

File: src/field_player_avatar.c [hunk 10]
```diff
@@ -0,0 +1,609 @@
...
+#include "metatile_behavior.h"
+#include "metatile_behaviors.h"
+#include "constants/flags.h"
+#include "event_data.h"
+#include "fieldmap.h"
+#include "overworld.h"
+#include "rotating_gate.h"
+#include "constants/event_object_movement_constants.h"
+
+bool8 ForcedMovement_None(void);
```
[the remark this answers is not in the document]
Comment (atasro2) [reply src/field_player_avatar.c]: forgot to include event_object_movement.h oops

File: src/field_player_avatar.c (hunk 9 above)
[the remark this answers is not in the document]
Comment (Diegoisawesome) [reply src/field_player_avatar.c]: Try reusing direction as a return var, or create a new return var. It might be enough to shuffle everything around.

**Pull request merged**